includes/class-wcpdf-admin.php

```diff
@@ -215,46 +215,129 @@ public function add_listing_actions( $order ) {
 	}
 	
 	/**
-	 * Create additional Shop Order column for Invoice Numbers
+	 * Create additional Shop Order column for Invoice Number/Date
 	 * @param array $columns shop order columns
 	 */
-	public function add_invoice_number_column( $columns ) {
+	public function add_invoice_columns( $columns ) {
 		// get invoice settings
-		$invoice = wcpdf_get_invoice( null );
+		$invoice          = wcpdf_get_invoice( null );
 		$invoice_settings = $invoice->get_settings();
-		if ( !isset( $invoice_settings['invoice_number_column'] ) ) {
-			return $columns;
+		$invoice_columns  = array(
+			'invoice_number_column' => __( 'Invoice Number', 'woocommerce-pdf-invoices-packing-slips' ),
+			'invoice_date_column'   => __( 'Invoice Date', 'woocommerce-pdf-invoices-packing-slips' ),
+		);
+
+		$offset = 2; // after order number column
+		foreach ( $invoice_columns as $slug => $name ) {
+			if ( ! isset( $invoice_settings[$slug] ) ) {
+				continue;
+			}
+
+			$columns = array_slice( $columns, 0, $offset, true ) +
+				array( $slug => $name ) +
+				array_slice( $columns, 2, count( $columns ) - 1, true ) ;
+
+			$offset++;
 		}
 
-		// put the column after the Status column
-		$new_columns = array_slice($columns, 0, 2, true) +
-			array( 'pdf_invoice_number' => __( 'Invoice Number', 'woocommerce-pdf-invoices-packing-slips' ) ) +
-			array_slice($columns, 2, count($columns) - 1, true) ;
-		return $new_columns;
+		return $columns;
 	}
 
 	/**
-	 * Display Invoice Number in Shop Order column (if available)
+	 * Display Invoice Number/Date in Shop Order column (if available)
 	 * @param  string $column column slug
 	 */
-	public function invoice_number_column_data( $column ) {
+	public function invoice_columns_data( $column ) {
 		global $post, $the_order;
 
-		if ( $column == 'pdf_invoice_number' ) {
-			$this->disable_storing_document_settings();
-			if ( empty( $the_order ) || WCX_Order::get_id( $the_order ) != $post->ID ) {
-				$order = WCX::get_order( $post->ID );
-				if ( $invoice = wcpdf_get_invoice( $order ) ) {
-					echo $invoice->get_number();
-				}
+		$this->disable_storing_document_settings();
+
+		$order = '';
+		if ( empty( $the_order ) || WCX_Order::get_id( $the_order ) != $post->ID ) {
+			$order = WCX::get_order( $post->ID );
+		} else {
+			$order = $the_order;
+		}
+
+		$invoice = wcpdf_get_invoice( $order );
+
+		if ( empty( $invoice ) ) {
+			return;
+		}
+
+		switch ( $column ) {
+			case 'invoice_number_column':
+				echo $invoice->get_number();
 				do_action( 'wcpdf_invoice_number_column_end', $order );
-			} else {
-				if ( $invoice = wcpdf_get_invoice( $the_order ) ) {
-					echo $invoice->get_number();
+				break;
+			case 'invoice_date_column':
+				if ( ! empty( $date = $invoice->get_date() ) ) {
+					echo $date->date_i18n( wcpdf_date_format() );
```

Comment:
@alexmigf can this be filtered to display the time too?

Reply:
Yes https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/e354f5cce95ea2feb28f1a84cbddb25a8176c886/includes/wcpdf-functions.php#L249

Reply:
But it can be limited to this context somehow?

Reply:
Done here [93d183f](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/309/commits/93d183f4eb8f5b038d95845d9a6c41b35d5b96c9)